de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/writer/WriterNodeModel.java

```diff
@@ -165,51 +199,26 @@ public static void writeFSKObject(FskPortObject fskObj, CombineArchive archive,
     // Adds model metadata
     addMetaData(archive, fskObj.modelMetadata, filePrefix + "metaData.json");
 
-    // If the model has an associated working directory with resources these resources
+ // If the model has an associated working directory with resources these resources
     // need to be saved into the archive.
     if (fskObj.getEnvironmentManager().isPresent()) {
       Optional<Path> workingDirectory = fskObj.getEnvironmentManager().get().getEnvironment();
       if (workingDirectory.isPresent()) {
         // Adds resources
         
         List<Path> resources = Files.list(workingDirectory.get()).collect(Collectors.toList());
```

Comment:
There is a leak here. The stream opened with Files.list is never closed. This can be dealt with a try-with resources:

```java
      try (Stream<Path> stream = Files.list(fskObj.getGeneratedResourcesDirectory().get().toPath())) {
        List<Path> resources = stream.collect(Collectors.toList());
        addResourcesToArchive(resources, archive, filePrefix, URIS);
      }
```